weave/ops_primitives/boolean.py

```diff
@@ -28,8 +28,14 @@ def bool_not(bool: bool) -> bool:  # type: ignore
 
 types.Boolean.instance_class = Boolean
 
-
-@op(name="none-coalesce")
-def none_coalesce(a: typing.Any, b: typing.Any) -> typing.Any:
-    # TODO: This logic is really complicated in Weavae0.
-    return a or b
+# Sort of made this work, but nowhere near the complexity of the js
+# implementation.
+# TODO: improve more?
+def none_coalesce_output_type(input_types):
+    return types.union(input_types['lhs'], input_types['rhs'])
+
+@op(name="none-coalesce",
+    output_type=none_coalesce_output_type)
+def none_coalesce(lhs: typing.Any, rhs: typing.Any):
```

Comment:
The complexity in WeaveJS has to do with nullability and mappability and ends up creating a bunch of different permutation options. It might be the case that the way it is used in your use case does not need more sophistication than what is here.